### PR TITLE
Issue #4940: fix twelfth part idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4940

All violations from the issue are fixed, except MultipleReturnPointsPerMethod, which will be addressed separately.

### CODE REVIEW

File: config/intellij-idea-inspections.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
     <inspection_tool class="AnonymousClassVariableHidesContainingMethodVariable" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="AnonymousFunctionJS" enabled="true" level="ERROR" enabled_by_default="true" />
     <!-- suppressed till GSoC project completion .... -->
```
Comment (romani) [inline config/intellij-idea-inspections.xml]: why comment is left ?

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 2]
```diff
@@ -133,7 +133,6 @@ public Checker createChecker(Configuration moduleConfig)
      * @param moduleConfig {@link Configuration} instance.
      * @return {@link Checker} instance.
      * @throws CheckstyleException if an exception occurs during checker configuration.
-     * @noinspection BooleanParameter
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: strange ..... is it required ?

Comment (Nimfadora) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: It is redundant, and my IDEA shows it as a redundant, why don't we remove it

File: config/intellij-idea-inspections.xml (hunk 1 above)
Comment (Nimfadora) [inline config/intellij-idea-inspections.xml]: Fixed

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 2 above)
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: Why is it redundant when we were required to make this change in test?
I even made a comment about this in https://github.com/checkstyle/checkstyle/issues/4857#issuecomment-320938777 .

Comment (romani) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: I reopened issue 4857 to resolve item in IT area.
I ok to remove this `noinspection` if TC is green, without it.

**Pull request merged**